File: testdata/fixtures/kotlin/CWE-79.kt
// CWE-79: XSS via WebView
package fixture
import android.webkit.WebView
class Test79 {
    fun vuln(view: WebView, user: String) {
        // ruleid: kotlin-xss
        view.loadData(user, "text/html", null)
        // ruleid: kotlin-xss
        view.loadDataWithBaseURL(null, user, "text/html", null, null)
    }
    fun safe(view: WebView) {
        // ok: kotlin-xss
        view.loadData("<p>hello</p>", "text/html", null)
        // ok: kotlin-xss
        view.loadDataWithBaseURL(null, "<b>fixed</b>", "text/html", null, null)
    }
    fun vulnExtra(view: WebView, input: String) {
        // ruleid: kotlin-xss
        view.loadData(input, "text/html", "utf-8")
    }
    fun vulnBase(view: WebView, html: String) {
        // ruleid: kotlin-xss
        view.loadDataWithBaseURL("https://example.com", html, "text/html", "utf-8", null)
    }
    fun safeLiteral(view: WebView) {
        // ok: kotlin-xss
        view.loadData("<html><body>safe</body></html>", "text/html", null)
    }
}

File: testdata/fixtures/c/CWE-79.c
/* CWE-79: XSS via non-literal output */
#include <stdio.h>
void vuln(const char *user) {
    char buf[256];
    /* ruleid: c-xss */
    printf(user);
    /* ruleid: c-xss */
    puts(user);
    /* ruleid: c-xss */
    sprintf(buf, "<div>%s</div>", user);
    /* ruleid: c-xss */
    fprintf(stdout, user);
    /* ok: c-xss */
    printf("<div>hello</div>");
    /* ok: c-xss */
    puts("safe");
    /* ok: c-xss */
    sprintf(buf, "fixed %s", "safe");
}
